contracts/amm/AmmTreasury.sol

```diff
@@ -131,6 +133,10 @@ contract AmmTreasury is
         IERC20Upgradeable(_asset).safeIncreaseAllowance(spender, Constants.MAX_VALUE);
     }
 
+    function getImplementation() external view override returns (address) {
```

Comment:
Maybe better is use that method in abstract contract to not repeat this in every contract